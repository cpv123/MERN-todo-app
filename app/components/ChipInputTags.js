import React, { Component } from 'react';
import PropTypes from 'prop-types';
import TextField from 'material-ui/TextField';
import ChipInput from 'material-ui-chip-input';

const ChipInputTags = (props) => {
    return (
        <ChipInput
            style = {{ 'marginRight': 15, 'marginBottom': 12.5 }}
            fullWidth = {true}
            floatingLabelText = "Tags"
            value={props.value}
            onRequestAdd={(chip) => props.onAddChip(chip)}
            onRequestDelete={(chip, index) => props.onRemoveChip(chip, index)}
        />
    );
}

ChipInputTags.propTypes = {
    floatingLabelText: PropTypes.string,
    value: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
    onRequestAdd: PropTypes.func.isRequired,
    onRequestDelete: PropTypes.func.isRequired
}

export default ChipInputTags;