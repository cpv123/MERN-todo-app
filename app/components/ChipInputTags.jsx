import React from 'react';
import PropTypes from 'prop-types';
import ChipInput from 'material-ui-chip-input';

const ChipInputTags = ({ value, onAddChip, onRemoveChip }) => (
  <ChipInput
    style={{ marginRight: 15, marginBottom: 12.5 }}
    fullWidth={false}
    floatingLabelText="Tags"
    value={value}
    onRequestAdd={chip => onAddChip(chip)}
    onRequestDelete={(chip, index) => onRemoveChip(chip, index)}
  />
);

ChipInputTags.propTypes = {
  value: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  onAddChip: PropTypes.func.isRequired,
  onRemoveChip: PropTypes.func.isRequired,
};

export default ChipInputTags;
