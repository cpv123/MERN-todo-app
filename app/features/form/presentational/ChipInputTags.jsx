import React from 'react';
import PropTypes from 'prop-types';
import ChipInput from 'material-ui-chip-input';

export default function ChipInputTags(props) {
  const { value, onAddChip, onRemoveChip } = props;
  return (
    <ChipInput
      style={{ marginRight: 15, marginBottom: 12.5 }}
      fullWidth={false}
      floatingLabelText="Tags"
      value={value}
      onRequestAdd={chip => onAddChip(chip)}
      onRequestDelete={(chip, index) => onRemoveChip(chip, index)}
    />
  );
}

ChipInputTags.propTypes = {
  value: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  onAddChip: PropTypes.func.isRequired,
  onRemoveChip: PropTypes.func.isRequired,
};
