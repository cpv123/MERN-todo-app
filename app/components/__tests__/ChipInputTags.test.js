/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import ChipInputTags from '../ChipInputTags.jsx';

describe('the ChipInputTags component', () => {
  it('renders correctly', () => {
    const mockProps = { 
      value: ["chipOne", "chip2"], 
      onAddChip: jest.fn(), 
      onRemoveChip: jest.fn() 
    };
    const wrapper = shallow(<ChipInputTags {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});