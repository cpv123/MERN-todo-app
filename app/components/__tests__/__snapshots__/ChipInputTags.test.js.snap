// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the ChipInputTags component renders correctly 1`] = `
<ChipInput
  allowDuplicates={false}
  clearOnBlur={true}
  filter={[Function]}
  floatingLabelText="Tags"
  fullWidth={false}
  newChipKeyCodes={
    Array [
      13,
    ]
  }
  onRequestAdd={[Function]}
  onRequestDelete={[Function]}
  style={
    Object {
      "marginBottom": 12.5,
      "marginRight": 15,
    }
  }
  underlineShow={true}
  value={
    Array [
      "chipOne",
      "chip2",
    ]
  }
/>
`;
